cpp/terrain/terrain.cpp

```diff
@@ -46,16 +47,15 @@ Terrain::Terrain(AssetManager &assetmanager,
 
 	this->terrain_id_count         = terrain_meta.size();
 	this->blendmode_count          = blending_meta.size();
-	this->textures                 = new Texture*[this->terrain_id_count];
-	this->blending_masks           = new Texture*[this->blendmode_count];
-	this->terrain_id_priority_map  = new int[this->terrain_id_count];
-	this->terrain_id_blendmode_map = new int[this->terrain_id_count];
-	this->influences_buf           = new struct influence[this->terrain_id_count];
+	this->textures.reserve(this->terrain_id_count);
+	this->blending_masks.reserve(this->blendmode_count);
+	this->terrain_id_priority_map  = util::make_unique<int[]>(this->terrain_id_count);
+	this->terrain_id_blendmode_map = util::make_unique<int[]>(this->terrain_id_count);
+	this->influences_buf           = util::make_unique<struct influence[]>(this->terrain_id_count);
```

Comment:
Why don't use a std::vector instead of unique_ptrs to arrays. Those vectors can be passed to all functions, that use them in a readonly way, as a const reference. Therefore unnecessary copies will be avoided.